### PR TITLE
Fix: qualify alter table table refs in optimizer qualify

### PR DESCRIPTION
**Pull request by z3z1ma**
As per discussion. We should qualify these table references if passed that particular expression type.

### CODE REVIEW

File: sqlglot/optimizer/qualify_tables.py [hunk 1]
```diff
@@ -46,6 +46,15 @@ def qualify_tables(
     db = exp.parse_identifier(db, dialect=dialect) if db else None
     catalog = exp.parse_identifier(catalog, dialect=dialect) if catalog else None
 
+    if isinstance(expression, exp.AlterTable):
+        for reference in expression.find_all(exp.Table):
+            if isinstance(reference.this, exp.Identifier):
+                if not reference.args.get("db"):
+                    reference.set("db", db)
+                if not reference.args.get("catalog") and reference.args.get("db"):
+                    reference.set("catalog", catalog)
```
Comment (georgesittas) [inline sqlglot/optimizer/qualify_tables.py:55]: Should we factor this section out? it's the same as in L78-82.

**Review by georgesittas — APPROVED** [review]
LGTM, just one comment.

File: sqlglot/optimizer/qualify_tables.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/optimizer/qualify_tables.py:55]: this is not the right approach

Comment (tobymao) [reply sqlglot/optimizer/qualify_tables.py:55]: i'm going to factor this out and use walk

Comment (tobymao) [reply sqlglot/optimizer/qualify_tables.py:55]: @z3z1ma can you change it to do this instead

```python

    for table in expression.find_all(exp.Table):
        if isinstance(table.this, exp.Identifier):
            if not table.args.get("db"):
                table.set("db", db)
            if not table.args.get("catalog") and table.args.get("db"):
                table.set("catalog", catalog)
```

delete the other sections

Comment (tobymao) [reply sqlglot/optimizer/qualify_tables.py:55]: move your test into qualify_tables.sql

File: sqlglot/optimizer/qualify_tables.py [hunk 2]
```diff
@@ -46,6 +46,18 @@ def qualify_tables(
...
+        if isinstance(table.this, exp.Identifier):
+            if not table.args.get("db"):
+                table.set("db", db)
+            if not table.args.get("catalog") and table.args.get("db"):
+                table.set("catalog", catalog)
+
+    if not isinstance(expression, exp.Subqueryable):
+        for table in expression.find_all(exp.Table):
+            _qualify(table)
+        return t.cast(E, expression)
```
Comment (tobymao) [inline sqlglot/optimizer/qualify_tables.py:59]: we don't want to do an early return here

Comment (tobymao) [inline sqlglot/optimizer/qualify_tables.py:59]:
```suggestion
```

**Review by tobymao — APPROVED** [review]

**Pull request merged**